src/node.cc

```diff
@@ -4265,35 +4265,17 @@ void Init(int* argc,
 }
 
 
-struct AtExitCallback {
-  AtExitCallback* next_;
-  void (*cb_)(void* arg);
-  void* arg_;
-};
-
-static AtExitCallback* at_exit_functions_;
-
-
-// TODO(bnoordhuis) Turn into per-context event.
 void RunAtExit(Environment* env) {
-  AtExitCallback* p = at_exit_functions_;
-  at_exit_functions_ = nullptr;
-
-  while (p) {
-    AtExitCallback* q = p->next_;
-    p->cb_(p->arg_);
-    delete p;
-    p = q;
-  }
+  env->RunAtExit();
 }
 
 
+thread_local Environment* thread_local_env;
```

Comment:
@addaleax @bnoordhuis I'd like to make sure I understand the reason for making this `static`. My understanding is that a `thread_local` object is allocated when the thread begins and deallocated when the thread ends, so `static` in this case does not refer to storage but to linkage. And without the explicit`static`, the linkage would be `external` by default. Does that sound correct?